### PR TITLE
fix(core): address verification should fail for uppercase bech32 addresses

### PR DESCRIPTION
**Pull request by 0xJacobV**
1. If an upper case bech32 address is provided, the isValidAddress function call should return false.
2. Apply es lint fixes to some files
3. Replaced mocha.opts with .mocharc.yml to remove deprecation warning

Issue: [BG-21499](https://bitgoinc.atlassian.net/browse/BG-21499)

### CODE REVIEW

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (tylerlevine) [inline modules/core/test/v2/unit/coins/ltc.ts:75]: This is a more idiomatic way to make sure a function throws a given error string.

```suggestion
      (() => ltc.canonicalAddress(bech32Address)).should.throw('invalid address');
```

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (tylerlevine) [inline modules/core/test/v2/unit/coins/ltc.ts:115]: Let's flip these back to not break existing behavior (see top level review comment)

**Review by tylerlevine — CHANGES_REQUESTED** [review]
I checked [BIP 173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki) which actually does cover the reasoning behind lowercase/uppercase bech32 addresses:

> The lowercase form is used when determining a character's value for checksum purposes.
\
Encoders MUST always output an all lowercase Bech32 string. If an uppercase version of the encoding result is desired, (e.g.- for presentation purposes, or QR code use), then an uppercasing procedure can be performed external to the encoding process.
\
Decoders MUST NOT accept strings where some characters are uppercase and some are lowercase (such strings are referred to as mixed case strings).
\
For presentation, lowercase is usually preferable, but inside QR codes uppercase SHOULD be used, as those permit the use of alphanumeric mode, which is 45% more compact than the normal byte mode. 

I think we should stick to the spec and continue to accept uppercase bech32 addresses as valid.

The issue that's causing problems for customers is that when we go to validate a transaction prebuild, we deserialize the transaction hex and look at the addresses corresponding to each output to see if they match the ones provided in the transaction build request.

That verification comparison happens inside a function called `verifyAddress` [here](https://github.com/BitGo/BitGoJS/blob/c8e9c566310b9f31cc43380a42b283d801d15b3f/modules/core/src/v2/coins/abstractUtxoCoin.ts#L924).

What I think we really ought to do is modify `verifyAddress` to check to see if the `addressType` is `p2wsh`, and if so, convert the address to lowercase before comparing to the expected address. Remember though, mixed case addresses should still be invalid according to the spec, so please make sure you include a test case for that.

The rest of the cleanup looks pretty good to me. Next time though, a separate PR for something tangental like the `mocha.opts` cleanup would be appreciated. Whitespace/formatting/comma/import fixes are acceptable as drive-by fixes, but please try to limit it to nothing more involved than that.

**Comment by tylerlevine** [comment]
Please see updated comment - your direction is correct but please see other review comments. You'll also need to rebase on https://github.com/BitGo/BitGoJS/pull/1353 once that gets merged to fix the build issues

**Comment by 0xJacobV** [comment]
@tylerlevine I made updates to the test description to reflect non-lowercase bech32 addresses. I don't think the implementation needs to change to account for it.

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (0xJacobV) [reply modules/core/test/v2/unit/coins/ltc.ts]: Suggestion applied

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (sangaman) [inline modules/core/test/v2/unit/coins/ltc.ts:67]: Might make sense to put some detail about how we only accept "canonical" lower case bech32 addresses, even though they are "valid" according to the spec, akin to the old comment. I know that if I came across this line without any context I'd be confused since I know upper case bech32 addresses to be valid according to the spec.

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (sangaman) [inline modules/core/test/v2/unit/coins/ltc.ts:80]: "upper case" rather than "non-lowercase"? Since we're explicitly testing upper case here, whereas non-lower could be mixed case as well (which is always invalid, maybe should be its own separate test case even).

**Review by sangaman — DISMISSED** [review]
LGTM, just a couple thoughts on comments/test naming. And I agree that it'd be helpful to put the style changes in a separate PR (tho it's good they're at least separate commits for history purposes) just to make reviewing easier.

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (0xJacobV) [reply modules/core/test/v2/unit/coins/ltc.ts]: @sangaman good idea. test description has been updated

File: modules/core/test/v2/unit/coins/ltc.ts
Comment (0xJacobV) [reply modules/core/test/v2/unit/coins/ltc.ts]: added more test cases to cover lowercase, uppercase and mixed case scenarios

**Review by OttoAllmendinger — APPROVED** [review]
please get a review from @CapnMigraine or @argjv for the stellar bump

**Review by tylerlevine — DISMISSED** [review]
nice work

**Review by OttoAllmendinger — APPROVED** [review]

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**
